airflow/providers/amazon/aws/hooks/elasticache_replication_group.py

```diff
@@ -0,0 +1,276 @@
+        """
+        Call ElastiCache API for creating a replication group
+
+        :param config: Configuration for creating the replication group
+        :type config: dict
+        :return: Response from ElastiCache create replication group API
+        :rtype: dict
+        """
+        return self.conn.create_replication_group(**config)
+
+    def delete_replication_group(self, replication_group_id):
+        """
+        Call ElastiCache API for deleting a replication group
+
+        :param replication_group_id: ID of replication group to delete
+        :type replication_group_id: str
+        :return: Response from ElastiCache delete replication group API
+        :rtype: dict
+        """
+        return self.conn.delete_replication_group(ReplicationGroupId=replication_group_id)
+
+    def describe_replication_group(self, replication_group_id):
+        """
+        Call ElastiCache API for describing a replication group
+
+        :param replication_group_id: ID of replication group to describe
+        :type replication_group_id: str
+        :return: Response from ElastiCache describe replication group API
+        :rtype: dict
+        """
+        return self.conn.describe_replication_groups(ReplicationGroupId=replication_group_id)
+
+    def get_replication_group_status(self, replication_group_id):
+        """
+        Get current status of replication group
+
+        :param replication_group_id: ID of replication group to check for status
+        :type replication_group_id: str
+        :return: Current status of replication group
+        :rtype: str
+        """
+        return self.describe_replication_group(replication_group_id)['ReplicationGroups'][0]['Status']
+
+    def is_replication_group_available(self, replication_group_id):
+        """
+        Helper for checking if replication group is available or not
+
+        :param replication_group_id: ID of replication group to check for availability
+        :type replication_group_id: str
+        :return: True if available else False
+        :rtype: bool
+        """
+        return self.get_replication_group_status(replication_group_id) == 'available'
+
+    def wait_for_availability(
+        self,
+        replication_group_id,
+        initial_sleep_time=None,
+        exponential_back_off_factor=None,
+        max_retries=None
+    ):
+        """
+        Check if replication group is available or not by performing a describe over it
+
+        :param max_retries: Max retries for checking availability of replication group
+        :type max_retries: int
+        :param exponential_back_off_factor: Factor for deciding next sleep time
+        :type exponential_back_off_factor: float
+        :param initial_sleep_time: Initial sleep time in seconds
+        :type initial_sleep_time: float
+        :param replication_group_id: ID of replication group to check for availability
+        :type replication_group_id: str
```

Comment:
Can you have doc order match function signature order.

And mention that back_off and sleep_time default to class level values if not passed.